sdk/eventhub/event-hubs/src/partitionLoadBalancer.ts

```diff
@@ -117,58 +117,61 @@ export class FairPartitionLoadBalancer implements PartitionLoadBalancer {
     return maxList[Math.floor(Math.random() * maxList.length)].partitionId;
   }
 
-  /*
-   * This method is called after determining that the load is not balanced. This method will evaluate
-   * if the current event processor should own more partitions. Specifically, this method returns true if the
-   * current event processor owns less than the minimum number of partitions or if it owns the minimum number
-   * and no other event processor owns lesser number of partitions than this event processor.
-   */
+  
   private _shouldOwnMorePartitions(
-    ourOwnerId: string,
-    minPartitionsPerEventProcessor: number,
-    partitionIds: string[],
-    ownerPartitionMap: Map<string, PartitionOwnership[]>
+    minRequired: number,
+    numEventProcessorsWithAdditionalPartition: number,
+    numPartitionsOwnedByUs: number,
+    processorCounts: ProcessorCounts
   ): boolean {
-    const numberOfPartitionsOwned = ownerPartitionMap.get(ourOwnerId)!.length;
+    let maxPartitionsPerProcessor = minRequired;
 
-    let sumOfPartitionsOwnedByAnyProcessor = 0;
-    for (const ownershipList of ownerPartitionMap.values()) {
-      sumOfPartitionsOwnedByAnyProcessor =
-        sumOfPartitionsOwnedByAnyProcessor + ownershipList.length;
+    if (
+      numEventProcessorsWithAdditionalPartition > 0 &&
+      processorCounts.haveExtraPartition < numEventProcessorsWithAdditionalPartition
+    ) {
+      maxPartitionsPerProcessor = minRequired + 1;
     }
+
+    return numPartitionsOwnedByUs >= maxPartitionsPerProcessor;
```

Comment:
Isn't this saying that we return true if we own _more than_ or _equal_ number of partitions as our max allowed? I think that's the opposite of what `_shouldOwnMorePartitions` should return.

Reply:
Fixed!